src/native/libs/System.Native/pal_signal.c

```diff
@@ -313,6 +313,15 @@ static void* SignalHandlerLoop(void* arg)
     free(arg);
     assert(pipeFd >= 0);
 
+#if defined(__linux__) || defined(__FreeBSD__)
+    pthread_setname_np(pthread_self(), ".NET SigHandler");
+#endif
+
+#if defined(__APPLE__)
+    // on macOS, pthread_setname_np only works for the calling thread.
+    pthread_setname_np(".NET Signal Handler");
```

Comment:
Should we use same name on all OSes here?

Reply:
Fixed. Thank you.